code/game/say.dm

```diff
@@ -77,9 +77,15 @@ var/list/freqtoname = list(
 	if(isnull(range))
 		range = 7
 	var/rendered = render_speech(speech)
-	for(var/atom/movable/AM in get_hearers_in_view(range, src) | observers)
+	var/list/listeners = get_hearers_in_view(range, src)
+	if(speech.speaker.GhostsAlwaysHear())
+		listeners += observers
```

Comment:
use `|=` to prevent duplicate messages if a ghost is also in range

Reply:
👌